File: passenger-service/src/test/java/com/software/modsen/passengerservice/util/TestEntities.java
package com.software.modsen.passengerservice.util;

import com.software.modsen.passengerservice.dto.request.PassengerCreateRequest;
import com.software.modsen.passengerservice.dto.request.PassengerRatingRequest;
import com.software.modsen.passengerservice.dto.request.PassengerUpdateRequest;
import com.software.modsen.passengerservice.entity.Passenger;
import com.software.modsen.passengerservice.entity.PassengerRating;
import lombok.experimental.UtilityClass;

@UtilityClass
public class TestEntities {
    public static final Long PASSENGER_ID = 1L;
    public static final String PASSENGER_FIRSTNAME = "Джон";
    public static final String PASSENGER_SURNAME = "Уильямс";
    public static final String PASSENGER_PATRONYMIC = "Джонсон";
    public static final String PASSENGER_EMAIL = "test@example.com";
    public static final String PASSENGER_PHONE_NUMBER = "+375331234567";
    public static final String SECOND_PASSENGER_EMAIL = "example@test.com";
    public static final String SECOND_PASSENGER_PHONE_NUMBER = "+375337654321";
    public static final Integer PAGE_NUMBER = 0;
    public static final Integer PAGE_SIZE = 10;
    public static final String SORT_BY_ID = "id";
    public static final Integer EXPECTED_LIST_SIZE = 1;
    public static final Integer EXPECTED_PASSENGER_LIST_SIZE = 2;
    public static final Integer FIRST_INDEX = 0;
    public static final Integer WANTED_NUMBER_OF_INVOCATIONS = 1;
    public static final Double PASSENGER_RATING = 4.55;
    public static final Long SECOND_PASSENGER_ID = 2L;
    public static final String PASSENGER_BASE_URL = "http://localhost:8081/api/v1/passengers";
    public static final String PASSENGER_RATING_BASE_URL = "http://localhost:8081/api/v1/passenger-ratings";

    public Passenger getTestPassenger() {
        return Passenger.builder()
                .id(PASSENGER_ID)
                .email(PASSENGER_EMAIL)
                .phoneNumber(PASSENGER_PHONE_NUMBER)
                .restricted(false)
                .build();
    }

    public Passenger getSecondTestPassenger() {
        return Passenger.builder()
                .id(SECOND_PASSENGER_ID)
                .email(PASSENGER_EMAIL)
                .phoneNumber(PASSENGER_PHONE_NUMBER)
                .build();
    }

    public PassengerRating getTestPassengerRating() {
        return PassengerRating.builder()
                .passenger(getTestPassenger())
                .rating(PASSENGER_RATING)
                .build();
    }

    public Passenger getPassengerForIT() {
        return Passenger.builder()
                .email(PASSENGER_EMAIL)
                .phoneNumber(PASSENGER_PHONE_NUMBER)
                .build();
    }

    public Passenger getSecondPassengerForIT() {
        return Passenger.builder()
                .email(SECOND_PASSENGER_EMAIL)
                .phoneNumber(SECOND_PASSENGER_PHONE_NUMBER)
                .build();
    }

    public PassengerCreateRequest getPassengerCreateRequestForIT() {
        return PassengerCreateRequest.builder()
                .firstname(PASSENGER_FIRSTNAME)
                .surname(PASSENGER_SURNAME)
                .patronymic(PASSENGER_PATRONYMIC)
                .email(PASSENGER_EMAIL)
                .phoneNumber(PASSENGER_PHONE_NUMBER)
                .build();
    }

    public PassengerUpdateRequest getPassengerUpdateRequestForIT() {
        return PassengerUpdateRequest.builder()
                .firstname(PASSENGER_FIRSTNAME)
                .surname(PASSENGER_SURNAME)
                .patronymic(PASSENGER_PATRONYMIC)
                .email(SECOND_PASSENGER_EMAIL)
                .phoneNumber(PASSENGER_PHONE_NUMBER)
                .build();
    }

    public PassengerRating getPassengerRatingForIT() {
        return PassengerRating.builder()
                .rating(PASSENGER_RATING)
                .build();
    }

    public PassengerRatingRequest getPassengerRatingRequestForIT() {
        return PassengerRatingRequest.builder()
                .passengerId(PASSENGER_ID)
                .passengerRating(PASSENGER_RATING)
                .build();
    }
}

File: passenger-service/src/test/kotlin/com/software/modsen/passengerservice/util/TestEntities.kt
package com.software.modsen.passengerservice.util

import com.software.modsen.passengerservice.dto.request.PassengerChangeStatusRequest
import com.software.modsen.passengerservice.dto.request.PassengerCreateRequest
import com.software.modsen.passengerservice.dto.request.PassengerRatingRequest
import com.software.modsen.passengerservice.dto.request.PassengerUpdateRequest
import com.software.modsen.passengerservice.entity.Passenger
import com.software.modsen.passengerservice.entity.PassengerRating

object TestEntities {
    const val PASSENGER_ID: Long = 1L
    const val PASSENGER_FIRSTNAME: String = "Джон"
    const val PASSENGER_SURNAME: String = "Уильямс"
    const val PASSENGER_PATRONYMIC: String = "Джонсон"
    const val PASSENGER_EMAIL: String = "test@example.com"
    const val PASSENGER_PHONE_NUMBER: String = "+375331234567"
    const val SECOND_PASSENGER_EMAIL: String = "example@test.com"
    const val SECOND_PASSENGER_PHONE_NUMBER: String = "+375337654321"
    const val PAGE_NUMBER: Int = 0
    const val PAGE_SIZE: Int = 10
    const val SORT_BY_ID: String = "id"
    const val EXPECTED_LIST_SIZE: Int = 1
    const val EXPECTED_PASSENGER_LIST_SIZE: Int = 2
    const val FIRST_INDEX: Int = 0
    const val WANTED_NUMBER_OF_INVOCATIONS: Int = 1
    const val PASSENGER_RATING: Double = 4.55
    const val SECOND_PASSENGER_ID: Long = 2L
    const val PASSENGER_BASE_URL: String = "http://localhost:8081/api/v1/passengers"
    const val PASSENGER_RATING_BASE_URL: String = "http://localhost:8081/api/v1/passenger-ratings"

    fun getTestPassenger(): Passenger {
        return Passenger(
            id = PASSENGER_ID,
            firstname = PASSENGER_FIRSTNAME,
            surname = PASSENGER_SURNAME,
            patronymic = PASSENGER_PATRONYMIC,
            email = PASSENGER_EMAIL,
            phoneNumber = PASSENGER_PHONE_NUMBER,
            restricted = false,
            passengerRating = null
        )
    }

    fun getTestPassengerChangeStatusRequest(): PassengerChangeStatusRequest {
        return PassengerChangeStatusRequest(
            id = PASSENGER_ID,
            status = true
        )
    }

    fun getSecondTestPassenger(): Passenger {
        return Passenger(
            id = SECOND_PASSENGER_ID,
            firstname = PASSENGER_FIRSTNAME,
            surname = PASSENGER_SURNAME,
            patronymic = PASSENGER_PATRONYMIC,
            email = PASSENGER_EMAIL,
            phoneNumber = PASSENGER_PHONE_NUMBER,
            restricted = false,
            passengerRating = null
        )
    }

    fun getTestPassengerRating(): PassengerRating {
        return PassengerRating(
            id = null,
            passenger = getTestPassenger(),
            rating = PASSENGER_RATING
        )
    }

    fun getPassengerForIT(): Passenger {
        return Passenger(
            id = null,
            firstname = PASSENGER_FIRSTNAME,
            surname = PASSENGER_SURNAME,
            patronymic = PASSENGER_PATRONYMIC,
            email = PASSENGER_EMAIL,
            phoneNumber = PASSENGER_PHONE_NUMBER,
            restricted = false,
            passengerRating = null
        )
    }

    fun getSecondPassengerForIT(): Passenger {
        return Passenger(
            id = null,
            firstname = PASSENGER_FIRSTNAME,
            surname = PASSENGER_SURNAME,
            patronymic = PASSENGER_PATRONYMIC,
            email = SECOND_PASSENGER_EMAIL,
            phoneNumber = SECOND_PASSENGER_PHONE_NUMBER,
            restricted = false,
            passengerRating = null
        )
    }

    fun getPassengerCreateRequestForIT(): PassengerCreateRequest {
        return PassengerCreateRequest(
            firstname = PASSENGER_FIRSTNAME,
            surname = PASSENGER_SURNAME,
            patronymic = PASSENGER_PATRONYMIC,
            email = PASSENGER_EMAIL,
            phoneNumber = PASSENGER_PHONE_NUMBER
        )
    }

    fun getPassengerUpdateRequestForIT(id: Long?): PassengerUpdateRequest {
        return PassengerUpdateRequest(
            id = id,
            firstname = PASSENGER_FIRSTNAME,
            surname = PASSENGER_SURNAME,
            patronymic = PASSENGER_PATRONYMIC,
            email = SECOND_PASSENGER_EMAIL,
            phoneNumber = PASSENGER_PHONE_NUMBER
        )
    }

    fun getTestPassengerChangeStatusRequest(id: Long?): PassengerChangeStatusRequest {
        return PassengerChangeStatusRequest(
            id = id!!,
            status = true
        )
    }

    fun getPassengerRatingForIT(passenger: Passenger): PassengerRating {
        return PassengerRating(
            id = null,
            passenger = passenger,
            rating = PASSENGER_RATING
        )
    }

    fun getPassengerRatingRequestForIT(passengerId: Long?): PassengerRatingRequest {
        return PassengerRatingRequest(
            passengerId = passengerId!!,
            passengerRating = PASSENGER_RATING
        )
    }
}